engines/cine/anim.cpp

```diff
@@ -798,8 +816,34 @@ int loadSeq(const char *resourceName, int16 idx) {
  */
 int loadResource(const char *resourceName, int16 idx, int16 frameIndex) {
 	int result = -1; // Return an error by default
+
+	if (g_cine->getGameType() == Cine::GType_OS &&
+		g_cine->getPlatform() == Common::kPlatformDOS &&
+		g_sound->musicType() != MT_MT32 &&
+		(strstr(resourceName, ".SPL") || strstr(resourceName, ".H32")))
+	{
+		char base[20];
+		removeExtention(base, resourceName);
+
+		for (int i = 0; i < ARRAYSIZE(resNameMapping); i++) {
+			if (scumm_stricmp(base, resNameMapping[i].from) == 0) {
+				strncpy(base, resNameMapping[i].to, sizeof(base));
```

Comment:
This is incorrect. Either use `sizeof(base) - 1` or better use `Common::strlcpy()`

Reply:
Taken into account (strlcpy).